CONTRIBUTING.adoc

```diff
@@ -20,6 +20,53 @@ Thank you for your contributions! Please follow this process to submit a patch:
 NOTE: If you submit a new add-on, the add-on must be added as a sub-directory in the *_/add-ons/_* sub-directory of the repository.
 See the documentation for information about link:https://docs.openshift.org/latest/minishift/using/addons.html#add-ons-overview[file structure and content format].
 
+[[testing]]
+== Testing Add-ons
+
+This repository uses integration test suite from `minishift/minishift` repository to test the functionality of the add-ons.
+The test suite is written in Go and uses Gherkin language for definition of test sets.
+This attitude allows add-on contributors to write tests effectively using predefined understandable sentences, for example:
+
+```gherkin
+  Scenario: Applying the CORS add-on
+    Given Minishift has state "Running"
+     When executing "minishift addons apply cors" succeeds
+     Then stdout should contain "CORS is now allowed from any address"
+```
+
+For more information about Gherkin you can check its link:https://github.com/cucumber/cucumber/wiki/Gherkin[wiki page].
+More in-depth information about Minishift integration test suite can be found at respective part of link:https://docs.openshift.org/latest/minishift/contributing/developing.html#integration-tests[Minishift documentation].
+
+=== Structure of Tests
+Feature files which define the test cases for each individual addon are stored in `test/integration/features` directory.
+The feature files follow naming convention of `<addon-name>.feature` and begins with the tag `@<addon-name>`.
+
+Integration tests are implemented in `test/integration/integration_test.go` file which imports test suite from Minishift repository and can be used for additional step definitions.
+To make the test execution friendlier `make integration` target is defined in the `Makefile`.
+
+=== Prerequisites
+
+To run the test successfully you need to make sure that:
+
+- Minishift prerequisites are fulfilled, Minishift is able to start on the machine
+- recent link:[Go] distribution (>=1.8) is installed
+- link:https://github.com/golang/dep[dep package] is installed
+
+Also please download latest Minishift binary and put it into `testing/bin`.
+Alternatively you can later start the tests specifying custom path to binary with `MINISHIFT_BINARY` parameter.
+
+To get all the needed go packages run:
+```
+make vendor
+```
+
+=== Running Tests
```

Comment:
Here, I believe, we should write how to run your addon's test instead of running all features files since the contributor is more interested in his tests.
 
```
=== Running tests for your Addon

Once the [prerequisites](#prerequisites) are fulfilled, run following command to test your addon:
'''
make integration GODOG_OPTS="-tags <addon-name>
'''

Note: `make integration` will run all tests which include tests of other addons.
```

Reply:
To make running tests more intuitive locally, I would prefer following command:
```
make integration ADDON=<addon-name>
```
which will add `GODOG_OPTS="-tags $(ADDON)` under the hood.

if no `ADDON`, then run all tests.